llmfoundry/callbacks/model_gauntlet_callback.py

```diff
@@ -120,8 +126,11 @@ def compute_averages(self, logger_data: Logger):
                     results[key] = [val]
         return {k: sum(v) / len(v) for k, v in results.items()}
 
-    def eval_end(self, state: State, logger: Logger):
-        new_metrics = self.compute_averages(logger)
+    def eval_after_all(self, state: Optional[State], logger: Logger):
+        inmemorylogger = get_in_memory_logger(logger)
```

Comment:
@bmosaicml what's the reason we need to grep the logs from the InMemoryLogger rather than read the eval metrics out of the state directly?

Reply:
+1 this is not very robust and strongly prefer a better way here.

Reply:
I think you should use `state.eval_metric_values` here. It should contain a dictionary of all recently computed metrics, keyed by metric name. The one issue is it is not keyed by dataloader name, so if there are multiple dataloaders with the same metric there will be a collision, but we can patch in that change to enable this PR